exporters/stats_stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -0,0 +1,331 @@
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.stats.Aggregation;
+import io.opencensus.stats.AggregationData;
+import io.opencensus.stats.AggregationData.CountData;
+import io.opencensus.stats.AggregationData.DistributionData;
+import io.opencensus.stats.AggregationData.MeanData;
+import io.opencensus.stats.AggregationData.SumDataDouble;
+import io.opencensus.stats.AggregationData.SumDataLong;
+import io.opencensus.stats.BucketBoundaries;
+import io.opencensus.stats.Measure;
+import io.opencensus.stats.View;
+import io.opencensus.stats.View.AggregationWindow;
+import io.opencensus.stats.View.AggregationWindow.Cumulative;
+import io.opencensus.stats.ViewData;
+import io.opencensus.stats.ViewData.AggregationWindowData;
+import io.opencensus.stats.ViewData.AggregationWindowData.CumulativeData;
+import io.opencensus.stats.ViewData.AggregationWindowData.IntervalData;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagKey.TagKeyString;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.TagValue.TagValueString;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
+
+/**
+ * Util methods to convert OpenCensus Stats data models to StackDriver monitoring data models.
+ */
+final class StackdriverExportUtils {
+
+  // TODO(songya): do we want these constants to be customizable?
+  @VisibleForTesting static final String NAMESPACE = "custom_metrics";
+  @VisibleForTesting static final String LABEL_DESCRIPTION = "OpenCensus TagKey";
+
+  // Construct a MetricDescriptor using a View.
+  @Nullable
+  static MetricDescriptor createMetricDescriptor(View view, String projectId) {
+    if (!(view.getWindow() instanceof Cumulative)) {
+      // TODO(songya): Only Cumulative view will be exported to Stackdriver in v0.1
+      return null;
+    }
+
+    MetricDescriptor.Builder builder = MetricDescriptor.newBuilder();
+    String name = view.getName().asString();
+    builder.setName(createMetricDescriptorName(projectId, name));
+    builder.setType(createMetricType(name));
+    builder.setDescription(view.getDescription());
+    builder.setUnit(view.getMeasure().getUnit());
+    // TODO(songya): do we want a different display name?
+    builder.setDisplayName(NAMESPACE + "/" + name);
+    for (TagKey tagKey : view.getColumns()) {
+      builder.addLabels(createLabelDescriptor(tagKey));
+    }
+    builder.setMetricKind(createMetricKind(view.getWindow()));
+    builder.setValueType(createValueType(view.getAggregation(), view.getMeasure()));
+    return builder.build();
+  }
+
+  // Construct a name for a MetricDescriptor
+  // Name format refers to cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors
+  private static String createMetricDescriptorName(
+      String projectId, String viewName) {
+    return String.format(
+        "projects/%s/metricDescriptors/custom.googleapis.com%%2Fcensus%%2F%s%%2F%s",
```

Comment:
See here for name:
https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/create

return String.format(projects/%s, project_id) -> also you can cache this because it is constant

Reply:
Done.